### PR TITLE
add installed/pruned workflow status/flag

### PR DESCRIPTION
**Pull request by dwsutherland**
Sibling to https://github.com/cylc/cylc-uiserver/pull/212

(This one should go in first)

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.py` and
  `conda-environment.yml`.
- [x] Tests added, and is used at the UIS.
- [x] No change log entry required (invisible to users).
- [x] No documentation update required.

### CODE REVIEW

**Review by kinow — APPROVED** [review]

**Comment by kinow** [comment]
Adding new states for the UI, with new icons. Chose folder-plus and folder-remove, but these icons may change later; we can also change states here later but updating the Cylc UI branch will be very simple.

image

**Comment by dwsutherland** [comment]
That codecov patch failure is wrong:
image
image

I put a print statement at the line it's complaining about, and the new unit test covers it.

File: cylc/flow/data_store_mgr.py [hunk 1]
```diff
@@ -254,6 +254,11 @@ def apply_delta(key, delta, data):
                     continue
     # Prune data elements
     if hasattr(delta, 'pruned'):
+        # UIS flag to prune workflow, set externally.
+        if key == WORKFLOW:
+            if delta.HasField('pruned'):
+                data[WORKFLOW].pruned = True
+            return
```
Comment (kinow) [inline cylc/flow/data_store_mgr.py:261]: Maybe simplify?

```py
        if key == WORKFLOW and delta.HasField('pruned'):
            data[WORKFLOW].pruned = True
            return
```

File: tests/unit/test_data_store_mgr.py [hunk 2]
```diff
@@ -46,3 +57,30 @@ def test_parse_job_item():
...
+    data = deepcopy(DATA_TEMPLATE)
+
+    assert data[WORKFLOW].id != w_id
+    assert data[WORKFLOW].pruned is False
+
+    for field, sub_delta in delta.ListFields():
+        apply_delta(field.name, sub_delta, data)
+
+    assert data[WORKFLOW].id == w_id
+    assert data[WORKFLOW].pruned is True
```
Comment (kinow) [inline tests/unit/test_data_store_mgr.py:86]: :+1:

**Review by kinow — APPROVED** [review]

**Review by oliver-sanders — APPROVED** [review]
Tested with the UIS PR, all good.

**Comment by oliver-sanders** [comment]
(leaving as is to be merged with https://github.com/cylc/cylc-uiserver/pull/212)

**Pull request merged**